### PR TITLE
fixup example.sh like 2.3.0 main.sh

### PR DESCRIPTION
**Pull request by mstreuhofer**
__i_am_main_script logic fixed for example.sh
added double-sourced test

### CODE REVIEW

**Comment by zbeekman** [comment]
@mstreuhofer if you're finished, I propose I rebase #86 right onto this PR. What are your thoughts?

**Comment by zbeekman** [comment]
> i'm done in this PR.
> if everybody agrees to the changes you can rebase your branch on mine and just merge yours after. this PR could then be closed.

@mstreuhofer Can I just add my commits to this PR? Because I think I have a fix for the MacOS build failures.... then we can just close one PR... I know @kvz is off enjoying his weekend already, and I don't see any reason why anyone would object to your proposed changes

File: test/fixture/main-nocolor.stdio [hunk 1]
```diff
@@ -5,7 +5,6 @@ ACCPTST:STDIO_REPLACE_DATETIMES
 {datetime} UTC [     info] __dir: {root}
 {datetime} UTC [     info] __base: main
 {datetime} UTC [     info] OSTYPE: {OSTYPE}
-{datetime} UTC [     info] Script called directly? yes
```
Comment (zbeekman) [inline test/fixture/main-nocolor.stdio]: Thanks, sorry I missed that

Comment (mstreuhofer) [inline test/fixture/main-nocolor.stdio]: np. now let's see what the tests are saying once they are done.

Comment (zbeekman) [inline test/fixture/main-nocolor.stdio]: I think I may need to move lanyon to dev dependency

Comment (zbeekman) [inline test/fixture/main-nocolor.stdio]: it looks like yarn is still broken, and without mentioning lanyon explicitly in package.json `npm ...` fails epically... I put it in plain old dependencies, but it looks like @kvz puts them in dev dependencies... https://github.com/yarnpkg/yarn/issues/760 is still unresolved, it appears

**Comment by zbeekman** [comment]
I'm going to make this change and then cancel the jobs, except for the latest push and pr job

**Comment by mstreuhofer** [comment]
ok. i'll stay clear until the dust settled.
good luck.

**Comment by zbeekman** [comment]
hmmmm I think https://github.com/kvz/bash3boilerplate/pull/87/commits/9c747d29b7d8907e839791c59779cec3dc59724d is not the fix... it's already listed in the dev dependencies... The issue may be that lanyon requires bundler which may be missing on OS X.... and of course the OS X CI queues are super backed up as always.

**Comment by zbeekman** [comment]
It looks like @kvz might be tweaking/fixing lanyon to resolve the OS X build failure... because I'm pretty flummoxed.

Save us, @kvz! lol

**Comment by kvz** [comment]
Trying @zbeekman! With a bunch of commits to master - for which I'm sorry, but I don't have a better way of debugging things on Travis that pass locally. (setting up a Travis VM locally is a bit beyond what I can invest atm).

**Comment by kvz** [comment]
Okay should be good @mstreuhofer @zbeekman (meaning: it's failing on this now: https://travis-ci.org/kvz/bash3boilerplate/jobs/201536450#L2779) - sorry for the delay!

**Comment by zbeekman** [comment]
thanks @kvz! Looking good. Not sure what that build failure was about, but the problem seems to have gone away... probably you were on a stale branch?

**Pull request merged**